### PR TITLE
Support xcode11

### PR DESCRIPTION
**Pull request by antrix1989**
Common core [PR](https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/535).

### CODE REVIEW

File: MSAL/MSAL.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -170,6 +168,9 @@
 		23A68A8320F538DE0071E435 /* MSALADFSAuthority.m in Sources */ = {isa = PBXBuildFile; fileRef = 23A68A7F20F538DE0071E435 /* MSALADFSAuthority.m */; };
 		23A68A9320F59E6B0071E435 /* NSString+MSALTestUtil.m in Sources */ = {isa = PBXBuildFile; fileRef = 23A68A9220F59E6B0071E435 /* NSString+MSALTestUtil.m */; };
 		23A68A9420F59E6B0071E435 /* NSString+MSALTestUtil.m in Sources */ = {isa = PBXBuildFile; fileRef = 23A68A9220F59E6B0071E435 /* NSString+MSALTestUtil.m */; };
+		23B1D35F22EA4798000954AF /* MSALPublicClientApplicationConfig.m in Sources */ = {isa = PBXBuildFile; fileRef = 23B1D35D22EA4797000954AF /* MSALPublicClientApplicationConfig.m */; };
+		23B1D36A22EA6E2F000954AF /* MSALPublicClientApplicationConfig.m in Sources */ = {isa = PBXBuildFile; fileRef = 23B1D35D22EA4797000954AF /* MSALPublicClientApplicationConfig.m */; };
+		23B1E4D222D4418000EE40BB /* MSALTestAppAcquireTokenViewController.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 23B1E4CB22D4418000EE40BB /* MSALTestAppAcquireTokenViewController.storyboard */; };
```
Comment (oldalton) [inline MSAL/MSAL.xcodeproj/project.pbxproj]: I assume this one is included in the test app target?

Comment (antrix1989) [reply MSAL/MSAL.xcodeproj/project.pbxproj]: yep

File: MSAL/MSAL.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -150,8 +150,10 @@
...
 		2364C74C1FB3E5CC00835428 /* XCTestCase+HelperMethods.m in Sources */ = {isa = PBXBuildFile; fileRef = 2364C7471FB3E52E00835428 /* XCTestCase+HelperMethods.m */; };
-		238BA014227BCAED00A5BACD /* MSALTenantProfile.h in Headers */ = {isa = PBXBuildFile; fileRef = 6077D49F22498BFF001798A2 /* MSALTenantProfile.h */; settings = {ATTRIBUTES = (Public, ); }; };
-		238BA01B227BCAEE00A5BACD /* MSALTenantProfile.h in Headers */ = {isa = PBXBuildFile; fileRef = 6077D49F22498BFF001798A2 /* MSALTenantProfile.h */; settings = {ATTRIBUTES = (Public, ); }; };
+		238A362D22EA3EB700F08167 /* MSALWebviewParameters.h in Headers */ = {isa = PBXBuildFile; fileRef = 2338294D22D7DC9E001B8AD6 /* MSALWebviewParameters.h */; settings = {ATTRIBUTES = (Public, ); }; };
+		238A362E22EA3ED400F08167 /* MSALWebviewParameters.h in Headers */ = {isa = PBXBuildFile; fileRef = 2338294D22D7DC9E001B8AD6 /* MSALWebviewParameters.h */; settings = {ATTRIBUTES = (Public, ); }; };
+		238BA014227BCAED00A5BACD /* MSALTenantProfile.h in Headers */ = {isa = PBXBuildFile; fileRef = 6077D49F22498BFF001798A2 /* MSALTenantProfile.h */; };
```
Comment (oldalton) [inline MSAL/MSAL.xcodeproj/project.pbxproj]: why is this not public and double?

File: MSAL/src/MSALPublicClientApplication.m [hunk 3]
```diff
@@ -644,18 +660,37 @@ - (void)acquireTokenForScopes:(NSArray<NSString *> *)scopes
...
     
     // This is meant to avoid developer error, when they configure PCA with e.g. AAD authority, but pass B2C authority here
     // Authority type in PCA and parameters should match
     if (![self.msalOauth2Provider isSupportedAuthority:requestAuthority])
     {
-        if (completionBlock)
+        if (block)
         {
             NSError *msidError = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidDeveloperParameter, @"Unsupported authority type. Please configure MSALPublicClientApplication with the same authority type", nil, nil, nil, nil, nil);
             NSError *msalError = [MSALErrorConverter msalErrorFromMsidError:msidError];
```
Comment (oldalton) [inline MSAL/src/MSALPublicClientApplication.m]: block already does conversion above

File: MSAL/src/MSALPublicClientApplication.m [hunk 4]
```diff
@@ -644,18 +660,37 @@ - (void)acquireTokenForScopes:(NSArray<NSString *> *)scopes
 - (void)acquireTokenSilentWithParameters:(MSALSilentTokenParameters *)parameters
                          completionBlock:(MSALCompletionBlock)completionBlock
 {
+    __auto_type block = ^(MSALResult *result, NSError *msidError, id<MSIDRequestContext> context)
+    {
+        NSError *msalError = [MSALErrorConverter msalErrorFromMsidError:msidError classifyErrors:YES msalOauth2Provider:self.msalOauth2Provider];
```
Comment (oldalton) [inline MSAL/src/MSALPublicClientApplication.m]: should we check that msidError is present before calling conversion?

File: MSAL/src/MSALPublicClientApplication.m [hunk 5]
```diff
@@ -683,10 +718,10 @@ - (void)acquireTokenSilentWithParameters:(MSALSilentTokenParameters *)parameters
     {
         MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Encountered an error when updating authority: %ld, %@", (long)authorityError.code, authorityError.domain);
         
-        if (completionBlock)
+        if (block)
         {
             NSError *msalError = [MSALErrorConverter msalErrorFromMsidError:authorityError];
```
Comment (oldalton) [inline MSAL/src/MSALPublicClientApplication.m]: block already does conversion above

Comment (antrix1989) [reply MSAL/src/MSALPublicClientApplication.m]: done

File: MSAL/src/MSALPublicClientApplication.m (hunk 3 above)
Comment (antrix1989) [reply MSAL/src/MSALPublicClientApplication.m]: done

File: MSAL/MSAL.xcodeproj/project.pbxproj (hunk 2 above)
Comment (antrix1989) [reply MSAL/MSAL.xcodeproj/project.pbxproj]: It is public, removed the duplicate row.

File: MSAL/src/MSALPublicClientApplication.m (hunk 4 above)
Comment (antrix1989) [reply MSAL/src/MSALPublicClientApplication.m]: msalErrorFromMsidError:msidError returns nil when we provide nil error to it, so this is not required.

**Pull request merged**